In [1]:
import pandas as pd
import matplotlib.pyplot as plt

importing the dataset

In [2]:
ipl = pd.read_csv("Updated_IPL_Matches_2008-2020.csv", parse_dates=["date"])
ipl = ipl.drop("Unnamed: 0",axis=1)
ipl.head()

id        city       date player_of_match  \
0  335982   Bangalore 2008-04-18     BB McCullum   
1  335983  Chandigarh 2008-04-19      MEK Hussey   
2  335984       Delhi 2008-04-19     MF Maharoof   
3  335985      Mumbai 2008-04-20      MV Boucher   
4  335986     Kolkata 2008-04-20       DJ Hussey   

                                               venue  neutral_venue  \
0                   M.Chinnaswamy Stadium, Bangalore              0   
1  Punjab Cricket Association IS Bindra Stadium, ...              0   
2                            Feroz Shah Kotla, Delhi              0   
3                           Wankhede Stadium, Mumbai              0   
4                              Eden Gardens, Kolkata              0   

                         team1                        team2  \
0  Royal Challengers Bangalore        Kolkata Knight Riders   
1              Kings XI Punjab          Chennai Super Kings   
2             Delhi Daredevils             Rajasthan Royals   
3               Mumbai Indians  Royal Challengers Bangalore   
4        Kolkata Knight Riders              Deccan Chargers   

                   toss_winner toss_decision                       winner  \
0  Royal Challengers Bangalore         field        Kolkata Knight Riders   
1          Chennai Super Kings           bat          Chennai Super Kings   
2             Rajasthan Royals           bat             Delhi Daredevils   
3               Mumbai Indians           bat  Royal Challengers Bangalore   
4              Deccan Chargers           bat        Kolkata Knight Riders   

    result  result_margin eliminator method    umpire1         umpire2  
0     runs          140.0          N    NaN  Asad Rauf     RE Koertzen  
1     runs           33.0          N    NaN  MR Benson      SL Shastri  
2  wickets            9.0          N    NaN  Aleem Dar  GA Pratapkumar  
3  wickets            5.0          N    NaN   SJ Davis       DJ Harper  
4  wickets            5.0          N    NaN  BF Bowden     K Hariharan

using the info method to get the insights.

In [3]:
ipl.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 816 entries, 0 to 815
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   id               816 non-null    int64         
 1   city             803 non-null    object        
 2   date             816 non-null    datetime64[ns]
 3   player_of_match  812 non-null    object        
 4   venue            816 non-null    object        
 5   neutral_venue    816 non-null    int64         
 6   team1            816 non-null    object        
 7   team2            816 non-null    object        
 8   toss_winner      816 non-null    object        
 9   toss_decision    816 non-null    object        
 10  winner           812 non-null    object        
 11  result           812 non-null    object        
 12  result_margin    799 non-null    float64       
 13  eliminator       812 non-null    object        
 14  method           19 non-null     object   

In this case, the most important columns that we will have to analyse are team1, team2, toss_winner, toss_decision and winner </br>
First let us look at the stats of all the teams combined

### 1. The total number of incidents where the toss winning team has also won the match

In [4]:
(ipl["toss_winner"] == ipl["winner"]).value_counts()

True     418
False    398
dtype: int64

So out of a possible 816 toss winners only 418 have actually gone on to win the match

In [5]:
(ipl["toss_winner"] == ipl["winner"]).value_counts(normalize=True) * 100

True     51.22549
False    48.77451
dtype: float64

This means that even if a team wins the toss, only there's a 51.22% chance that it will win the match

### 2. Now let's look at these stats team wise

In [6]:
ipl_toss_actual_winner = ipl[(ipl["toss_winner"] == ipl["winner"])]
ipl_toss_actual_winner[["toss_winner", "winner"]]

toss_winner                       winner
1            Chennai Super Kings          Chennai Super Kings
8               Rajasthan Royals             Rajasthan Royals
10              Rajasthan Royals             Rajasthan Royals
12               Deccan Chargers              Deccan Chargers
14           Chennai Super Kings          Chennai Super Kings
..                           ...                          ...
804  Royal Challengers Bangalore  Royal Challengers Bangalore
806          Sunrisers Hyderabad          Sunrisers Hyderabad
807               Delhi Capitals               Delhi Capitals
813          Sunrisers Hyderabad          Sunrisers Hyderabad
814               Delhi Capitals               Delhi Capitals

[418 rows x 2 columns]

In [7]:
ipl_toss_actual_winner["winner"].value_counts()

Mumbai Indians                 61
Chennai Super Kings            61
Kolkata Knight Riders          55
Rajasthan Royals               44
Royal Challengers Bangalore    43
Kings XI Punjab                36
Delhi Daredevils               35
Sunrisers Hyderabad            29
Deccan Chargers                19
Gujarat Lions                  10
Delhi Capitals                 10
Rising Pune Supergiant          8
Kochi Tuskers Kerala            4
Pune Warriors                   3
Name: winner, dtype: int64

As we can see Chennai Super Kings and Mumbai Indians are the two teams, </br> which have gone on to win the match after winning the toss most times

### 3. Now let's look at each team individually

#### 1. Chennai Super Kings

In [8]:
#Selecting all the rows where either team1 is CSK or team2 is CSK
team1_csk = ipl["team1"] == "Chennai Super Kings"
team2_csk = ipl["team2"] == "Chennai Super Kings"
ipl_csk = ipl[team1_csk | team2_csk]

In [9]:
len(ipl_csk)

178

This means that Chennai have played a total number of 178 matches between 2008 and 2020

In [10]:
(ipl_csk["winner"] == "Chennai Super Kings").value_counts()

True     106
False     72
Name: winner, dtype: int64

This means that Chennai have won 106 matches and have lost 72 matches overall

In [11]:
(ipl_csk["toss_winner"] == "Chennai Super Kings").value_counts()

True     97
False    81
Name: toss_winner, dtype: int64

This means that Chennai have won the toss 97 times overall

In [12]:
win_csk = ipl_csk["winner"] == "Chennai Super Kings"
toss_csk  = ipl_csk["toss_winner"] == "Chennai Super Kings"
both_csk = ipl_csk[win_csk & toss_csk]
len(both_csk)

61

This means that Chennai have won the game 61 times after winning the toss as well

In [13]:
not_win_csk = ipl_csk["winner"] != "Chennai Super Kings"
toss_csk  = ipl_csk["toss_winner"] == "Chennai Super Kings"
len(ipl_csk[not_win_csk & toss_csk])

36

This means that Chennai lost the match 36 times even after winning the toss

In [85]:
win_rate = len(both_csk)/(len(both_csk)+len(ipl_csk[not_win_csk & toss_csk])) * 100
win_rate

62.88659793814433

This means csk won 62.8 percent of the matches after winning the toss

#### 2. Mumbai Indians

In [14]:
#Selecting all the rows where either team1 is MI or team2 is MI
team1_mi = ipl["team1"] == "Mumbai Indians"
team2_mi = ipl["team2"] == "Mumbai Indians"
ipl_mi = ipl[team1_mi | team2_mi]

In [15]:
len(ipl_mi)

203

This means that Mumbai have played a total number of 203 matches between 2008 and 2020

In [16]:
(ipl_mi["winner"] == "Mumbai Indians").value_counts()

True     120
False     83
Name: winner, dtype: int64

This means that Mumbai have won 120 matches and have lost 83 matches overall

In [17]:
(ipl_mi["toss_winner"] == "Mumbai Indians").value_counts()

True     106
False     97
Name: toss_winner, dtype: int64

This means that Mumbai have won the toss 106 times overall

In [18]:
win_mi = ipl_mi["winner"] == "Mumbai Indians"
toss_mi  = ipl_mi["toss_winner"] == "Mumbai Indians"
both_mi = ipl_mi[win_mi & toss_mi]
len(both_mi)

61

This means that Mumbai have won the game 61 times after winning the toss as well

In [19]:
not_win_mi = ipl_mi["winner"] != "Mumbai Indians"
toss_mi  = ipl_mi["toss_winner"] == "Mumbai Indians"
len(ipl_mi[not_win_mi & toss_mi])

45

This means that Mumbai lost the match 45 times even after winning the toss

In [86]:
win_rate = len(both_mi)/(len(both_mi)+len(ipl_mi[not_win_mi & toss_mi])) * 100
win_rate

57.54716981132076

#### 3. Kolkata Knight Riders

In [20]:
#Selecting all the rows where either team1 is KKR or team2 is KKR
team1_kkr = ipl["team1"] == "Kolkata Knight Riders"
team2_kkr = ipl["team2"] == "Kolkata Knight Riders"
ipl_kkr = ipl[team1_kkr | team2_kkr]

In [21]:
len(ipl_kkr)

192

This means that Kolkata have played a total number of 192 matches between 2008 and 2020

In [22]:
(ipl_kkr["winner"] == "Kolkata Knight Riders").value_counts()

True     99
False    93
Name: winner, dtype: int64

This means that Kolkata have won 99 matches and have lost 93 matches overall

In [23]:
(ipl_kkr["toss_winner"] == "Kolkata Knight Riders").value_counts()

True     98
False    94
Name: toss_winner, dtype: int64

This means that Kolkata have won the toss 98 times overall

In [24]:
win_kkr = ipl_kkr["winner"] == "Kolkata Knight Riders"
toss_kkr  = ipl_kkr["toss_winner"] == "Kolkata Knight Riders"
both_kkr = ipl_kkr[win_kkr & toss_kkr]
len(both_kkr)

55

This means that Kolkata have won the game 55 times after winning the toss as well

In [25]:
not_win_kkr = ipl_kkr["winner"] != "Kolkata Knight Riders"
toss_kkr  = ipl_kkr["toss_winner"] == "Kolkata Knight Riders"
len(ipl_kkr[not_win_kkr & toss_kkr])

43

This means that Kolkata lost the match 43 times even after winning the toss

In [87]:
win_rate = len(both_kkr)/(len(both_kkr)+len(ipl_kkr[not_win_kkr & toss_kkr])) * 100
win_rate

56.12244897959183

#### 4. Rajasthan Royals

In [26]:
#Selecting all the rows where either team1 is RR or team2 is RR
team1_rr = ipl["team1"] == "Rajasthan Royals"
team2_rr = ipl["team2"] == "Rajasthan Royals"
ipl_rr = ipl[team1_rr | team2_rr]

In [27]:
len(ipl_rr)

161

This means that Rajasthan have played a total number of 161 matches between 2008 and 2020

In [28]:
(ipl_rr["winner"] == "Rajasthan Royals").value_counts()

True     81
False    80
Name: winner, dtype: int64

This means that Rajasthan have won 81 matches and have lost 80 matches overall

In [29]:
(ipl_rr["toss_winner"] == "Rajasthan Royals").value_counts()

True     87
False    74
Name: toss_winner, dtype: int64

This means that Rajasthan have won the toss 87 times overall </br>
This also means that Rajasthan have won the toss more times than winning a match (81)

In [30]:
win_rr = ipl_rr["winner"] == "Rajasthan Royals"
toss_rr  = ipl_rr["toss_winner"] == "Rajasthan Royals"
both_rr = ipl_rr[win_rr & toss_rr]
len(both_rr)

44

This means that Rajasthan have won the game 44 times after winning the toss as well

In [31]:
not_win_rr = ipl_rr["winner"] != "Rajasthan Royals"
toss_rr  = ipl_rr["toss_winner"] == "Rajasthan Royals"
len(ipl_rr[not_win_rr & toss_rr])

43

This means that Rajasthan lost the match 43 times even after winning the toss

In [88]:
win_rate = len(both_rr)/(len(both_rr)+len(ipl_rr[not_win_rr & toss_rr])) * 100
win_rate

50.57471264367817

#### 5. Royal Challengers Bangalore

In [32]:
#Selecting all the rows where either team1 is RCB or team2 is RCB
team1_rcb = ipl["team1"] == "Royal Challengers Bangalore"
team2_rcb = ipl["team2"] == "Royal Challengers Bangalore"
ipl_rcb = ipl[team1_rcb | team2_rcb]

In [33]:
len(ipl_rcb)

195

This means that Bangalore have played a total number of 195 matches between 2008 and 2020

In [34]:
(ipl_rcb["winner"] == "Royal Challengers Bangalore").value_counts()

False    104
True      91
Name: winner, dtype: int64

This means that Bangalore have won 91 matches and have lost 104 matches overall </br>
This also means that they have lost more matches than they have won

In [35]:
(ipl_rcb["toss_winner"] == "Royal Challengers Bangalore").value_counts()

False    108
True      87
Name: toss_winner, dtype: int64

This means that Bangalore have won the toss 87 times overall

In [36]:
win_rcb = ipl_rcb["winner"] == "Royal Challengers Bangalore"
toss_rcb  = ipl_rcb["toss_winner"] == "Royal Challengers Bangalore"
both_rcb = ipl_rcb[win_rcb & toss_rcb]
len(both_rcb)

43

This means that Bangalore have won the game 43 times after winning the toss as well

In [37]:
not_win_rcb = ipl_rcb["winner"] != "Royal Challengers Bangalore"
toss_rcb  = ipl_rcb["toss_winner"] == "Royal Challengers Bangalore"
len(ipl_rcb[not_win_rcb & toss_rcb])

44

This means that Bangalore lost the match 44 times even after winning the toss <br>
This is more than the number of times (43), they have won the game times after winning the toss

In [89]:
win_rate = len(both_rcb)/(len(both_rcb)+len(ipl_rcb[not_win_rcb & toss_rcb])) * 100
win_rate

49.42528735632184

#### 6. Kings XI Punjab

In [38]:
#Selecting all the rows where either team1 is KXIP or team2 is KXIP
team1_kxip = ipl["team1"] == "Kings XI Punjab"
team2_kxip = ipl["team2"] == "Kings XI Punjab"
ipl_kxip = ipl[team1_kxip | team2_kxip]

In [39]:
len(ipl_kxip)

190

This means that Punjab have played a total number of 190 matches between 2008 and 2020

In [40]:
(ipl_kxip["winner"] == "Kings XI Punjab").value_counts()

False    102
True      88
Name: winner, dtype: int64

This means that Punjab have won 88 matches and have lost 102 matches overall </br>
This also means that they have lost more matches than they have won

In [41]:
(ipl_kxip["toss_winner"] == "Kings XI Punjab").value_counts()

False    105
True      85
Name: toss_winner, dtype: int64

This means that Punjab have won the toss 85 times overall

In [42]:
win_kxip = ipl_kxip["winner"] == "Kings XI Punjab"
toss_kxip  = ipl_kxip["toss_winner"] == "Kings XI Punjab"
both_kxip = ipl_kxip[win_kxip & toss_kxip]
len(both_kxip)

36

This means that Punjab have won the game 36 times after winning the toss as well

In [43]:
not_win_kxip = ipl_kxip["winner"] != "Kings XI Punjab"
toss_kxip  = ipl_kxip["toss_winner"] == "Kings XI Punjab"
len(ipl_kxip[not_win_kxip & toss_kxip])

49

This means that Punjab lost the match 49 times even after winning the toss </br>
This is more than the number of times (36), they have won the game times after winning the toss

In [90]:
win_rate = len(both_kxip)/(len(both_kxip)+len(ipl_kxip[not_win_kxip & toss_kxip])) * 100
win_rate

42.35294117647059

#### 7. Delhi Daredevils and Delhi Capitals

In [44]:
#Selecting all the rows where either team1 is DD or DC or team2 is DD or DC
team1_dd = ipl["team1"] == "Delhi Daredevils"
team2_dd = ipl["team2"] == "Delhi Daredevils"
team1_dc = ipl["team1"] == "Delhi Capitals"
team2_dc = ipl["team2"] == "Delhi Capitals"
ipl_del = ipl[team1_dd | team2_dd | team1_dc | team2_dc]

In [45]:
len(ipl_del)

194

This means that both the Delhi teams have played a combined total number of 194 matches between 2008 and 2020

In [46]:
ipl_del["winner"].isin(["Delhi Daredevils", "Delhi Capitals"]).value_counts()

False    108
True      86
Name: winner, dtype: int64

This means that Delhi teams have won a combined 86 matches and have lost 108 matches overall </br>
This also means that they have lost more matches than they have won

In [47]:
ipl_del["toss_winner"].isin(["Delhi Daredevils", "Delhi Capitals"]).value_counts()

True     100
False     94
Name: toss_winner, dtype: int64

This means that Delhi have won the toss 100 times overall </br>
This is also means that they have the toss more number of times than they have won matches (86)

In [48]:
win_del = ipl_del["winner"].isin(["Delhi Daredevils", "Delhi Capitals"])
toss_del  = ipl_del["toss_winner"].isin(["Delhi Daredevils", "Delhi Capitals"])
both_del = ipl_del[win_del & toss_del]
len(both_del)

45

This means that Delhi (both combined) have won the game 45 times after winning the toss as well

In [49]:
not_win_del = ~win_del
toss_del  = ipl_del["toss_winner"].isin(["Delhi Daredevils", "Delhi Capitals"])
len(ipl_del[not_win_del & toss_del])

55

This means that Delhi (both combined) lost the match 55 times even after winning the toss </br>
This is more than the number of times (45), they have won the game times after winning the toss

In [91]:
win_rate = len(both_del)/(len(both_del)+len(ipl_del[not_win_del & toss_del])) * 100
win_rate

45.0

#### 8. Sunrisers Hyderabad and Deccan Chargers

In [50]:
#Selecting all the rows where either team1 is SRH or DCH or team2 is SRH or DCH
team1_srh = ipl["team1"] == "Sunrisers Hyderabad"
team2_srh = ipl["team2"] == "Sunrisers Hyderabad"
team1_dch = ipl["team1"] == "Deccan Chargers"
team2_dch = ipl["team2"] == "Deccan Chargers"
ipl_hyd = ipl[team1_srh | team2_srh | team1_dch | team2_dch]

In [51]:
len(ipl_hyd)

199

This means that both the Hyderabad teams have played a combined total number of 199 matches between 2008 and 2020

In [52]:
ipl_hyd["winner"].isin(["Sunrisers Hyderabad", "Deccan Chargers"]).value_counts()

False    104
True      95
Name: winner, dtype: int64

This means that Hyderabad teams have won a combined 95 matches and have lost 104 matches overall </br>
This also means that they have lost more matches than they have won

In [53]:
ipl_hyd["toss_winner"].isin(["Sunrisers Hyderabad", "Deccan Chargers"]).value_counts()

True     100
False     99
Name: toss_winner, dtype: int64

This means that Hyderabad have won the toss 100 times overall </br>
This is also means that they have the toss more number of times than they have won matches (95)

In [54]:
win_hyd = ipl_hyd["winner"].isin(["Sunrisers Hyderabad", "Deccan Chargers"])
toss_hyd  = ipl_hyd["toss_winner"].isin(["Sunrisers Hyderabad", "Deccan Chargers"])
both_hyd = ipl_hyd[win_hyd & toss_hyd]
len(both_hyd)

48

This means that Hyderabad (both combined) have won the game 48 times after winning the toss as well

In [55]:
not_win_hyd = ~win_hyd
toss_hyd = ipl_hyd["toss_winner"].isin(["Sunrisers Hyderabad", "Deccan Chargers"])
len(ipl_hyd[not_win_hyd & toss_hyd])

52

This means that Hyderabad (both combined) lost the match 52 times even after winning the toss </br>
This is more than the number of times (48), they have won the game times after winning the toss

In [92]:
win_rate = len(both_hyd)/(len(both_hyd)+len(ipl_hyd[not_win_hyd & toss_hyd])) * 100
win_rate

48.0

#### 9. Gujarat Lions

In [56]:
#Selecting all the rows where either team1 is GL or team2 is GL
team1_gl = ipl["team1"] == "Gujarat Lions"
team2_gl = ipl["team2"] == "Gujarat Lions"
ipl_gl = ipl[team1_gl | team2_gl]

In [57]:
len(ipl_gl)

30

This means that both the Gujarat played a total number of 30 matches between 2008 and 2020

In [58]:
(ipl_gl["winner"] == "Gujarat Lions").value_counts()

False    17
True     13
Name: winner, dtype: int64

This means that Gujarat teams have won 13 matches and have lost 17 matches overall </br>
This also means that they have lost more matches than they have won

In [59]:
(ipl_gl["toss_winner"] == "Gujarat Lions").value_counts()

True     15
False    15
Name: toss_winner, dtype: int64

This means that Gujarat have won the toss 15 times overall </br>
This is also means that they have the toss more number of times than they have won matches (13)

In [60]:
win_gl = ipl_gl["winner"] == "Gujarat Lions"
toss_gl  = ipl_gl["toss_winner"] == "Gujarat Lions"
both_gl = ipl_gl[win_gl & toss_gl]
len(both_gl)

10

This means that Gujarat have won the game 10 times after winning the toss as well

In [61]:
not_win_gl = ipl_gl["winner"] != "Gujarat Lions"
toss_gl  = ipl_gl["toss_winner"] == "Gujarat Lions"
len(ipl_gl[not_win_gl & toss_gl])

5

This means that Gujarat lost the match 5 times even after winning the toss

In [93]:
win_rate = len(both_gl)/(len(both_gl)+len(ipl_gl[not_win_gl & toss_gl])) * 100
win_rate

66.66666666666666

#### 10. Rising Pune Supergiant and Pune Warriors      

In [62]:
#Selecting all the rows where either team1 is RPS or PW or team2 is RPS or PW
team1_rps = ipl["team1"] == "Rising Pune Supergiant"
team2_rps = ipl["team2"] == "Rising Pune Supergiant"
team1_pw = ipl["team1"] == "Pune Warriors"
team2_pw = ipl["team2"] == "Pune Warriors"
ipl_pun = ipl[team1_rps | team2_rps | team1_pw | team2_pw]

In [63]:
len(ipl_pun)

76

This means that both the Pune teams have played a combined total number of 76 matches between 2008 and 2020

In [64]:
ipl_pun["winner"].isin(["Rising Pune Supergiant", "Pune Warriors"]).value_counts()

False    49
True     27
Name: winner, dtype: int64

This means that Pune teams have won a combined 27 matches and have lost 49 matches overall </br>
This also means that they have lost more matches than they have won

In [65]:
ipl_pun["toss_winner"].isin(["Rising Pune Supergiant", "Pune Warriors"]).value_counts()

False    43
True     33
Name: toss_winner, dtype: int64

This means that Pune have won the toss 33 times overall </br>
This is also means that they have the toss more number of times than they have won matches (27)

In [66]:
win_pun = ipl_pun["winner"].isin(["Rising Pune Supergiant", "Pune Warriors"])
toss_pun  = ipl_pun["toss_winner"].isin(["Rising Pune Supergiant", "Pune Warriors"])
both_pun = ipl_pun[win_pun & toss_pun]
len(both_pun)

11

This means that Pune (both combined) have won the game 11 times after winning the toss as well

In [67]:
not_win_pun = ~win_pun
toss_pun  = ipl_pun["toss_winner"].isin(["Rising Pune Supergiant", "Pune Warriors"])
len(ipl_pun[not_win_pun & toss_pun])

22

This means that Pune (both combined) lost the match 22 times even after winning the toss </br>
This is more than the number of times (11), they have won the game times after winning the toss

In [94]:
win_rate = len(both_pun)/(len(both_pun)+len(ipl_pun[not_win_pun & toss_pun])) * 100
win_rate

33.33333333333333

#### 11. Kochi Tuskers Kerala

In [68]:
#Selecting all the rows where either team1 is KTK or team2 is KTK
team1_ktk = ipl["team1"] == "Kochi Tuskers Kerala"
team2_ktk = ipl["team2"] == "Kochi Tuskers Kerala"
ipl_ktk = ipl[team1_ktk | team2_ktk]

In [69]:
len(ipl_ktk)

14

This means that both the Kochi played a total number of 14 matches between 2008 and 2020

In [70]:
(ipl_ktk["winner"] == "Kochi Tuskers Kerala").value_counts()

False    8
True     6
Name: winner, dtype: int64

This means that Kochi teams have won 6 matches and have lost 8 matches overall </br>
This also means that they have lost more matches than they have won

In [71]:
(ipl_ktk["toss_winner"] == "Kochi Tuskers Kerala").value_counts()

True     8
False    6
Name: toss_winner, dtype: int64

This means that Gujarat have won the toss 8 times overall </br>
This is also means that they have the toss more number of times than they have won matches (6)

In [72]:
win_ktk = ipl_ktk["winner"] == "Kochi Tuskers Kerala"
toss_ktk  = ipl_ktk["toss_winner"] == "Kochi Tuskers Kerala"
both_ktk = ipl_ktk[win_ktk & toss_ktk]
len(both_ktk)

4

This means that Kochi have won the game 4 times after winning the toss as well

In [73]:
not_win_ktk = ipl_ktk["winner"] != "Kochi Tuskers Kerala"
toss_ktk  = ipl_ktk["toss_winner"] == "Kochi Tuskers Kerala"
len(ipl_ktk[not_win_ktk & toss_ktk])

4

This means that Kochi lost the match 4 times even after winning the toss